jena-arq/src/main/java/org/apache/jena/query/ResultSetFormatter.java

```diff
@@ -516,6 +520,41 @@ static public void outputAsJSON(boolean booleanResult)
     static public void outputAsJSON(OutputStream outStream, boolean booleanResult)
     { output(outStream, booleanResult, SPARQLResultSetJSON) ; }
     
+    /** Output an iterator of JSON values.
+    *
+    * @param outStream output stream
+    * @param jsonItems The JSON values
+    */
+   public static void outputAsJSON(OutputStream outStream, Iterator<JsonObject> jsonItems)
+   {
+       JSWriter jWriter = new JSWriter(outStream) ;
+       jWriter.startArray() ;
+       jWriter.startOutput() ;
+       while (jsonItems.hasNext()) 
+       {
+           jWriter.startObject() ;
+           JsonObject jsonItem = jsonItems.next() ;
+           for (Entry<String, JsonValue> entry: jsonItem.entrySet()) 
+           {
+               JsonValue value = entry.getValue() ;
+               String val = "";
+               if (value.isString()) {
```

Comment:
@afs but here I didn't have a `Node`, only the `JsonValue`, and had to convert from `JsonValue` to `String`

Reply:
"outputAsJSON" is a bit confusing because the other functions of the same name produce SPARQL results in JSON.  How about "output"?

In the code, numbers still come out as strings.

JSWriter seems to be only good for its current usage. The code below seems to work: it manages the outer array itself and uses `JsonValue.output` to write each item.

```
 public static void outputAsJSON(OutputStream outStream, Iterator<JsonObject> jsonItems)
    {
        IndentedWriter out = new IndentedWriter(outStream);
        out.println("[");
        out.incIndent();

        while (jsonItems.hasNext()) {
            JsonObject jsonItem = jsonItems.next() ;
            jsonItem.output(out);
            if ( jsonItems.hasNext() )
                out.println(" ,");
            else
                out.println();
        }
        out.decIndent();
        out.println("]");
        out.flush();
    }
```

Reply:
>"outputAsJSON" is a bit confusing because the other functions of the same name produce SPARQL results in JSON. How about "output"?

Sounds good to me. Done!

>In the code, numbers still come out as strings.

Oh, d'oh me! You raised the issue, I updated the code but not the part that writes it back to the user. And even added a test that was testing it was getting "123" and "true" in the output. Fixed the test, and thanks heaps for the code snippet! Otherwise I'd be scratching my head and thinking how to use JSWriter.

Updated, and uploaded the data below

```
@prefix :          <http://example.org/> .

:s :p 123 .
:s :p "abc" .
:s :p "def"@en .
```

Then queried `JSON { "s": ?s , "p": ?p , "o" : ?o } WHERE { ?s ?p ?o }`. Screen shot below with the `123` value - hopefully - correctly displayed! :man_dancing: :tada: 

![screenshot_2018-03-20_21-02-30](https://user-images.githubusercontent.com/304786/37642710-116538c6-2c83-11e8-95e4-d14a28e57a5a.png)